pyqtorch/time_dependent/solvers.py

```diff
@@ -30,7 +30,7 @@ def ode_fun(self, t: float, psi: Tensor) -> Tensor:
         with warnings.catch_warnings():
             # filter-out UserWarning about "Sparse CSR tensor support is in beta state"
             warnings.filterwarnings("ignore", category=UserWarning)
-            res = -1j * self.H(t) @ psi.to_sparse()
+            res = -1j * self.H(t) @ psi.to_sparse() if self.options.use_sparse else psi
```

Comment:
Just to make sure, I would use use parenthesis:
-1j * self.H(t) @ `(psi.to_sparse() if self.options.use_sparse else psi)`

Reply:
it was actually really needed :)